packages/workspace/src/command-line/affected.ts

```diff
@@ -46,38 +46,46 @@ export function affected(command: string, parsedArgs: yargs.Arguments): void {
   try {
     switch (command) {
       case 'apps':
-        const apps = affectedProjects
-          .filter((p) => p.type === ProjectType.app)
-          .map((p) => p.name);
-        if (parsedArgs.plain) {
-          console.log(apps.join(' '));
-        } else {
-          printArgsWarning(nxArgs);
-          if (apps.length) {
-            output.log({
-              title: 'Affected apps:',
-              bodyLines: apps.map((app) => `${output.colors.gray('-')} ${app}`),
-            });
-          }
-        }
-        break;
+      case 'libs': {
+        const projectTypeToFilter =
+          command === 'apps' ? ProjectType.app : ProjectType.lib;
+
+        const projects = affectedProjects.filter(
+          (p) => p.type === projectTypeToFilter
+        );
 
-      case 'libs':
-        const libs = affectedProjects
-          .filter((p) => p.type === ProjectType.lib)
-          .map((p) => p.name);
         if (parsedArgs.plain) {
-          console.log(libs.join(' '));
+          const affectedProjectsNames = projects.map((p) => p.name);
+          console.log(affectedProjectsNames.join(' '));
+        } else if (nxArgs.target) {
+          const libsWithTarget = allProjectsWithTarget(projects, nxArgs);
+
+          runCommand(
```

Comment:
Hm it is not the intended behavior for `nx affected:apps` or libs to actually run tasks... It should be purely informational. We can filter for the ones with tasks.. but I don't think the behavior should be changed to run tasks.

Reply:
I thought that you and I already discussed what I wanted to achieve in the issue #1830, even you said [_"This makes sense"_](https://github.com/nrwl/nx/issues/1830#issuecomment-562983753)

Did something change since the last time we spoke or am I missing something?

Reply:
Sorry, I must have glossed over this before :

> And nothing more happens. Should run the deploy architect of my libraries or apps

I don't think it should run tasks. I thought the expected behavior in that issue was to only print projects.